dags/blocksec_plugin/erc20_balanceof_operator.py

```diff
@@ -1,28 +1,18 @@
-from airflow.models.baseoperator import BaseOperator
-from airflow.utils.decorators import apply_defaults
-from blocksec_plugin.web3_hook import Web3Hook
 
-ERC20_ABI = '''
-[{"inputs":[{"internalType":"address","name":"account","type":"address"}],"name":"balanceOf","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"}]
-'''
+from airflow.utils.decorators import apply_defaults
+from blocksec_plugin.contract_interaction_operator import ContractInteractionOperator
+from blocksec_plugin.abis import ERC20_ABI_BALANCE
 
-class ERC20BalanceOfOperator(BaseOperator):
-    template_fields = []
+class ERC20BalanceOfOperator(ContractInteractionOperator):
+    template_fields = ["account"]
     ui_color = "#BC9EC1"
 
     @apply_defaults
     def __init__(self,
-                 contract_address,
                  account,
-                 web3_conn_id='web3_default',
                  *args,
                  **kwargs):
         super().__init__(*args, **kwargs)
         self.account = account
-        self.web3_conn_id = web3_conn_id
-        self.contract_address = contract_address
-        self.web3 = Web3Hook(web3_conn_id=self.web3_conn_id).http_client
-
-    def execute(self, context):
-        contract = self.web3.eth.contract(self.contract_address, abi=ERC20_ABI)
-        return contract.functions.balanceOf(self.account).call()
+        self.function_args = {"account": self.account}
+        self.function = self.contract.functions.balanceOf
```

Comment:
For this one, I think we need to override `execute` since calling `balanceOf` is a `view` method on the ERC20 contract, therefore, if it uses the parent execute, it'll have issues since you can't execute the balance of transaction